packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -51,12 +53,50 @@ public void onMethodCall(MethodCall call, final Result result) {
       case "StorageReference#getDownloadUrl":
         getDownloadUrl(call, result);
         break;
+      case "StorageReference#getMetadata":
+        getMetadata(call, result);
+        break;
       default:
         result.notImplemented();
         break;
     }
   }
 
+  private void getMetadata(MethodCall call, final Result result) {
+    String path = call.argument("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    ref.getMetadata()
+        .addOnSuccessListener(
+            new OnSuccessListener<StorageMetadata>() {
+              @Override
+              public void onSuccess(StorageMetadata storageMetadata) {
+                HashMap<String, Object> map = new HashMap();
+                map.put("name", storageMetadata.getName());
+                map.put("bucket", storageMetadata.getBucket());
+                map.put("generation", storageMetadata.getGeneration());
+                map.put("metadataGeneration", storageMetadata.getMetadataGeneration());
+                map.put("path", storageMetadata.getPath());
+                map.put("sizeBytes", storageMetadata.getSizeBytes());
+                map.put("creationTimeMillis", storageMetadata.getCreationTimeMillis());
+                map.put("updatedTimeMillis", storageMetadata.getUpdatedTimeMillis());
+                map.put("md5Hash", storageMetadata.getMd5Hash());
+                map.put("cacheControl", storageMetadata.getCacheControl());
+                map.put("contentDisposition", storageMetadata.getContentDisposition());
+                map.put("contentEncoding", storageMetadata.getContentEncoding());
+                map.put("contentLanguage", storageMetadata.getContentLanguage());
+                map.put("contentType", storageMetadata.getContentType());
+                result.success(map);
+              }
+            })
+        .addOnFailureListener(
+            new OnFailureListener() {
+              @Override
+              public void onFailure(@NonNull Exception e) {
```

Comment:
If we are to use `@NonNull`, we should do it consistently. I'd vote for not using nullability annotations at all, because they are not present in the Flutter APIs.

Reply:
It should to have the annotation since it is overriding a @Nonnull parameter. Will add it where it's missing.